lib/src/signing.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright 2023 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![allow(missing_docs)]
+
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::sync::RwLock;
+
+use thiserror::Error;
+
+use crate::backend::CommitId;
+use crate::settings::UserSettings;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum SigStatus {
+    /// Valid signature that matches the data.
+    Good,
+    /// Valid signature that could not be verified (e.g. due to an unknown key).
+    Unknown,
+    /// Valid signature that does not match the signed data.
+    Bad,
+    /// Invalid signature.
+    Invalid,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Verification {
+    /// The status of the signature.
+    pub status: SigStatus,
+    /// The key representation, if available. For GPG, this will be the key
+    /// fingerprint.
+    pub key: Option<String>,
+    /// A display string, if available. For GPG, this will be formatted primary
+    /// user ID.
+    pub display: Option<String>,
+}
```

Comment:
This type is kind of more generic because whatever you put in key/display is backend-specific
For example, gpg backend has `key` for `unknown` signatures, because the keyid is in the signature, but not `display` because duh, the key is unknown :)

Although SigStatus being a separate enum is historic, I recall there was a separate use for it in some of my earlier impl

Having a enum with 3 out of 4 variants having the same fields is inconvenient, hence the unit enum+struct, but I like idea to have some specific `VerificationError` variant for invalid signatures - would have to handle that one specially in the templater